x-pack/plugin/eql/src/main/antlr/EqlBase.g4

```diff
@@ -211,30 +201,22 @@ DECIMAL_VALUE
     ;
 
 IDENTIFIER
-    : (LETTER | '_') (LETTER | DIGIT | '_' | '@' )*
-    ;
-
-DIGIT_IDENTIFIER
-    : DIGIT (LETTER | DIGIT | '_' | '@')+
+    : (LETTER | '_') (LETTER | DIGIT | '_')*
     ;
 
-QUOTED_IDENTIFIER
```

Comment:
Without a quoted identifier declaration, query `where WHERE true` would not work. Replace `where` with any keyword or field name containing special characters (quotes, etc...).
Even if the current queries don't use special fields, quoting is mandatory for user defined indices.